MobileCenterUpdates/MobileCenterUpdates/MSUpdates.m

```diff
@@ -365,39 +362,80 @@ - (BOOL)isNewerVersion:(MSReleaseDetails *)details {
-
-  // Show the alert controller.
-  [alertController show];
+  // Displaying alert dialog. Running on main thread.
+  dispatch_async(dispatch_get_main_queue(), ^{
+
+    // TODO: The text should be localized. There is a separate task for resources.
+    NSString *releaseNotes =
+        details.releaseNotes ? details.releaseNotes : @"No release notes were provided for this release.";
+
+    MSAlertController *alertController =
+        [MSAlertController alertControllerWithTitle:@"Update available" message:releaseNotes];
+
+    // Add a "Ignore"-Button
+    [alertController addDefaultActionWithTitle:@"Ignore"
+                                       handler:^(UIAlertAction *action) {
+                                         MSLogDebug([MSUpdates logTag], @"Ignore the release id: %@.", details.id);
+                                         [MS_USER_DEFAULTS setObject:details.id forKey:kMSIgnoredReleaseIdKey];
+                                       }];
+
+    // Add a "Postpone"-Button
+    [alertController addCancelActionWithTitle:@"Postpone"
+                                      handler:^(UIAlertAction *action) {
+                                        MSLogDebug([MSUpdates logTag], @"Postpone the release for now.");
+                                      }];
+
+    // Add a "Download"-Button
+    [alertController addDefaultActionWithTitle:@"Download"
+                                       handler:^(UIAlertAction *action) {
+                                         MSLogDebug([MSUpdates logTag], @"Start download and install the release.");
+                                         [self startDownload:details];
+                                       }];
+
+    // Show the alert controller.
+    [alertController show];
+  });
 }
 
 // TODO: Please implement!
 - (void)startDownload:(MSReleaseDetails *)details {
 }
 
+- (void)openUrl:(NSURL *)url {
+  if ([self isEnabled]) {
+
+    // If the request is not for Mobile Center Updates, ignore.
+    if (![kMSUpdtsDefaultCustomScheme isEqualToString:[url scheme]]) {
+      return;
+    }
+
+    // Parse query parameters
+    NSString *requestedId = [MS_USER_DEFAULTS objectForKey:kMSUpdateTokenRequestIdKey];
+    NSString *queryRequestId = nil;
+    NSString *queryUpdateToken = nil;
+    NSURLComponents *components = [NSURLComponents componentsWithURL:url resolvingAgainstBaseURL:NO];
+
+    for (NSURLQueryItem *item in components.queryItems) {
+      if ([item.name isEqualToString:kMSUpdtsURLQueryRequestIdKey]) {
+        queryRequestId = item.value;
+      } else if ([item.name isEqualToString:kMSUpdtsURLQueryUpdateTokenKey]) {
+        queryUpdateToken = item.value;
+      }
+    }
+
+    // If the request ID doesn't match, ignore.
+    if (!(requestedId && queryRequestId && [requestedId isEqualToString:queryRequestId])) {
+      return;
+    }
+
+    // Store update token
+    if (queryUpdateToken) {
+      [MSKeychainUtil storeString:queryUpdateToken forKey:kMSUpdateTokenKey];
```

Comment:
Maybe add a log here saying the update token has been successfully retrieved. No need to print it I think or hide it as it's a secret.